invoice/src/main/resources/org/killbill/billing/invoice/dao/InvoiceSqlDao.sql.stg

```diff
@@ -95,61 +89,78 @@ getInvoiceByAccountRecordIdAfter() ::= <<
    <defaultOrderBy("")>
 >>
 
-searchByAccountOrInvoiceId(ordering) ::= <<
-(select
-<allTableFields("t.")>
-from <tableName()> t
-where (<idField("t.")> = :searchKey)
-<andCheckSoftDeletionWithComma("t.")>
-<AND_CHECK_TENANT("t.")>)
-UNION
-(select
-<allTableFields("t.")>
-from <tableName()> t
-where (t.account_id = :searchKey)
-<andCheckSoftDeletionWithComma("t.")>
-<AND_CHECK_TENANT("t.")>)
-order by <recordIdField("")> <ordering>
-limit :rowCount offset :offset
-;
->>
-
-getSearchByAccountOrInvoiceIdCount() ::= <<
-select count(*) from
-((select
-<allTableFields("t.")>
-from <tableName()> t
-where (<idField("t.")> = :searchKey)
-<andCheckSoftDeletionWithComma("t.")>
-<AND_CHECK_TENANT("t.")>)
-UNION
-(select
-<allTableFields("t.")>
-from <tableName()> t
-where (t.account_id = :searchKey)
-<andCheckSoftDeletionWithComma("t.")>
-<AND_CHECK_TENANT("t.")>)) as count
-;
+invoiceBalanceQuery() ::= <<
+SELECT
+      x.invoice_id
+    , x.tenant_record_id
+    , SUM(x.amount) as balance
+FROM (
+    SELECT
+          i.id as invoice_id
+        , i.tenant_record_id
+        , 0 as amount
+    FROM invoices i
```

Comment:
As discussed, we need to make sure:
- Invoice is Committed (not draft/void)
- Invoice is not migrated
- Invoice is not marked as written_off
- For HA, we need to look at the parent invoice

Reply:
I am resolving this, since this was addressed in https://github.com/killbill/killbill/pull/2016#issue-2320212379